web/components/ItemComponent.vue

```diff
@@ -0,0 +1,20 @@
+
+<script setup lang="ts">
+import { type Item,resolveComponent } from "#imports";
+
+const { item } = defineProps<{
+  item: Item
+}>()
+
+
+const myComponent = resolveComponent(item.type)
+</script>
+
+<template>
+  <div>
+    <component v-if="myComponent" :is="myComponent" :item="item" />
```

Comment:
There is another change in main branch that should be considered here. I want to use the same component in both list view and single item view, because they are going to reuse a lot of logic. Basically you just need to add isList prop.